### PR TITLE
Prefer implicit span conversions

### PR DESCRIPTION
**Pull request by jjonescz**
Corresponding speclet update: https://github.com/dotnet/csharplang/pull/8149
Test plan: https://github.com/dotnet/roslyn/issues/73445

### CODE REVIEW

**Review by cston — APPROVED** [review]

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs [hunk 1]
```diff
@@ -2320,6 +2320,87 @@ .locals init (object[] V_0)
...
+            {
+                public static void M(Span<int> x) => Console.Write(1);
+                public static void M(IEnumerable<int> x) => Console.Write(2);
+            }
+            """;
+
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular12).VerifyDiagnostics(
+            // (5,3): error CS0121: The call is ambiguous between the following methods or properties: 'C.M(Span<int>)' and 'C.M(IEnumerable<int>)'
+            // C.M(a);
+            Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("C.M(System.Span<int>)", "C.M(System.Collections.Generic.IEnumerable<int>)").WithLocation(5, 3));
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs:2343]: I'm not understanding why this is an error? `M(Span<int>)` should be an implicit span conversion, no?

**Review by 333fred — COMMENTED** [review]
Are there places that we updated to avoid ambiguities in the bootstrap build that should be put back?

File: src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs (hunk 1 above)
Comment (jjonescz) [reply src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs:2343]: Yes, exactly, that's why this error appears only in C# 12 and not in later versions.

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs (hunk 1 above)
Comment (333fred) [reply src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs:2368]: Ah, right. Sorry, I read too fast.

**Review by 333fred — APPROVED** [review]

**Pull request merged**